### PR TITLE
Removes global public docker registry variable.

### PR DESCRIPTION
**Pull request by zionyx**
Removes registry variable validation attribute to allow empty registry value.
Added more tests to test for empty registry parsed in.

### CODE REVIEW

File: Public/Invoke-DockerPush.ps1 [hunk 1]
```diff
@@ -4,7 +4,6 @@ function Invoke-DockerPush {
     [CmdletBinding()]
     param (
         [Parameter(ValueFromPipelineByPropertyName = $true)]
-        [ValidateNotNull()]
```
Comment (rasmusjelsgaard) [inline Public/Invoke-DockerPush.ps1]: Could you add a test case that tests what happens when you pass in a null value? 

As I understand it, Add-Postfix will throw an exception in that scenario.

File: Tests/Integration.Tests.ps1 [hunk 2]
```diff
@@ -85,8 +81,8 @@ Describe 'Use cases for this module' {
 
             # 2. Build and push image to latest tag, then grab it and see it's ok
             $result = Invoke-DockerBuild -Registry $localRegistryName -ImageName $imageName |
-                        Invoke-DockerPush |
-                            Invoke-DockerPull
+                Invoke-DockerPush |
```
[the remark this answers is not in the document]
Comment (zionyx) [reply Tests/Integration.Tests.ps1]: Beautify did this for me. 🌻🤣

File: Public/Invoke-DockerPull.ps1 [hunk 3]
```diff
@@ -1,11 +1,9 @@
 function Invoke-DockerPull {
     [CmdletBinding()]
     param (
-
         [Parameter(ValueFromPipelineByPropertyName = $true)]
-        [ValidateNotNullOrEmpty()]
         [String]
-        $Registry = $global:DockerPublicRegistry,
+        $Registry = '',
```
[the remark this answers is not in the document]
Comment (zionyx) [reply Public/Invoke-DockerPull.ps1]: Yup. Add-Postfix allows null / empty too.

File: Tests/Integration.Tests.ps1 [hunk 4]
```diff
@@ -27,18 +27,14 @@ Describe 'Use cases for this module' {
             Invoke-Command 'docker container rm -v registry'
         }
 
-        BeforeEach {
-            $script:backupLocation = Get-Location
```
[the remark this answers is not in the document]
Comment (zionyx) [reply Tests/Integration.Tests.ps1]: Reverted.

File: Public/Invoke-DockerPush.ps1 (hunk 1 above)
Comment (zionyx) [reply Public/Invoke-DockerPush.ps1]: Add-Postfix checks if it's null or white space:
https://github.com/3shapeAS/dockerbuild-pwsh/blob/master/Private/Utilities.ps1#L45L47

So i think we're safe. Will add more tests like you suggested.

Comment (zionyx) [reply Public/Invoke-DockerPush.ps1]: Added test in DockerBuild. `$null` will not break anything because it is overriden by the default value.

File: Private/Utilities.ps1 [hunk 5]
```diff
@@ -1,22 +1,16 @@
...
-    if ($IsWindows) {
-        return [system.io.path]::GetTempPath()
-    } elseif ($IsLinux) {
-        return '/tmp'
-    }
-}
-
 function New-RandomFolder {
+
+    $driveRoot = (Get-PSDrive TestDrive).Root
```
Comment (rasmusjelsgaard) [inline Private/Utilities.ps1]: Nice one.

Comment (zionyx) [reply Private/Utilities.ps1]: The credit goes to @casz :praisesun:

File: Public/Invoke-DockerPush.ps1 (hunk 1 above)
Comment (zionyx) [reply Public/Invoke-DockerPush.ps1]: Added `$null` value tests for it.

File: Public/Invoke-DockerPull.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (zionyx) [reply Public/Invoke-DockerPull.ps1]: `Add-Postfix` now trims whitespace value, and does it even if the trimmed value is empty.
Added `$null` and `whitespace` tests for it.

File: Tests/Invoke-DockerPull.Tests.ps1 [hunk 6]
```diff
@@ -45,14 +45,21 @@ Describe 'Pull docker images' {
...
+            Write-Debug $result
+            $result | Should -BeLikeExactly "docker pull ubuntu:bionic"
         }
 
         It 'pulls public docker image by image name and digest' {
             Invoke-DockerPull -ImageName 'ubuntu' -Digest 'sha256:a7b8b7b33e44b123d7f997bd4d3d0a59fafc63e203d17efedf09ff3f6f516152'
             $result = GetMockValue -Key "pull"
             Write-Debug $result
-            $result | Should -BeLikeExactly "docker pull ${global:DockerPublicRegistry}/ubuntu@sha256:a7b8b7b33e44b123d7f997bd4d3d0a59fafc63e203d17efedf09ff3f6f516152"
+            $result | Should -BeLikeExactly "docker pull ubuntu@sha256:a7b8b7b33e44b123d7f997bd4d3d0a59fafc63e203d17efedf09ff3f6f516152"
```
[the remark this answers is not in the document]
Comment (zionyx) [reply Tests/Invoke-DockerPull.Tests.ps1]: Sure but can i add it in a separate PR? This PR is getting crowded.

**Comment by zionyx** [comment]
Rebased to fix cherry pick caused merge conflict and force pushed.

**Comment by zionyx** [comment]
Fixed all outstanding issues! Approval please 🙌
